odoo/addons/base/models/ir_ui_view.py

```diff
@@ -590,11 +589,10 @@ def _get_inheriting_views(self):
         """
         if not self.ids:
             return self.browse()
-        self.browse().check_access('read')
         domain = self._get_inheriting_views_domain()
-        e = expression(domain, self.env['ir.ui.view'])
-        where_clause = e.query.where_clause
-        assert e.query.from_clause == SQL.identifier('ir_ui_view'), f"Unexpected from clause: {e.query.from_clause}"
+        query = self._search(domain)
+        where_clause = query.where_clause
```

Comment:
That is not the same behavior, right? Before we didn't take in account ir.rule and now we do. Is it intentional ? Or I miss something ?